### ISSUE TITLE
Request to keep oldest modification date when merging, or a choice of oldest,newest,etc.

### ISSUE THREAD

**Issue by pixelb** [pixelb/fslint#80]
Original [issue 81](https://code.google.com/p/fslint/issues/detail?id=81) created by pixelb on 2012-06-28T01:25:10.000Z:

Hi! Thanks for a very useful program!

Would it be possible to add a feature to select which modification time is kept for a group of files to be merged?

This would be very useful for old photos, videos, etc when there isn't any time/date stored in the file itself.

I've been gathering family photos and videos which have made it onto the hard drive in various ways, which means I get duplicates very often, but which have different modification times because the times were often not preserved when they were transferred.

I think it would make the most sense to default to keeping the oldest date found among the files as the one to keep when merging, since the oldest time is likely the closest to correct for when file was actually created (except maybe for when the date is for some reason set to the beginning of the Epoch in 1970, then maybe could choose to do the opposite).

In version 2.42, I know that we can select &quot;all but newest&quot; or &quot;all but oldest&quot;, but I can't find any info on what actually happens after that.  And, even if selected files are &quot;not&quot; merged, then it still doesn't solve the problem of the merged files possibly ending up with the wrong date.

What do you think about adding some kind of option to handle this?  Would it just need the file arguments in a certain order on the command line to end up with the intended date?

Thank you and have a great day! ^^

**Comment by sixtyfive** [comment]
For reasons beyond my control, I can't use either hardlinking nor delete duplicates straight out; instead I need to symlink to the *oldest* duplicate. So far that doesn't seem to be possible with fslint (it always symlinks to the *newest* duplicate. One thing I tried (out of an unreflected expectation for it to work) was to click the "Date" column header in the results. I thought that it's perhaps the first duplicate within each group that gets symlinked to. Anyways, perhaps that'd be an idea?

**Comment by pixelb** [comment]
Well when you "select all but oldest", the symlinking/hardlinking is just done within that selection. I.E the oldest file will not be linked to the group. Really "select all but ...." is only useful for subsequent deletion

We'd have to as you suggest support ordering by date, and then the link operation to always keep the first entry

**Comment by sixtyfive** [comment]
> Well when you "select all but oldest", the symlinking/hardlinking is just done within that selection. I.E the oldest file will not be linked to the group. Really "select all but ...." is only useful for subsequent deletion

Yes, I tried that to, and found out that it works like you describe.

> We'd have to as you suggest support ordering by date, and then the link operation to always keep the first entry

Is that something you'll do at some point?

**Comment by pixelb** [comment]
Yes but probably not soon. It's a bit involved

**Comment by endolith** [comment]
> That could be useful I suppose, but is a bit esoteric and
the first request I've had for that.

I use this feature *all* the time in AllDup.  I am almost always deleting all but oldest:

![2020-04-23 21_03_11-AllDup 4 4 22 - Search Result 'File content Byte by Byte (100%)'](https://user-images.githubusercontent.com/58611/80164045-eff30280-85a5-11ea-956d-a85e72d9e1c7.png)
